internal/profile/install_profile.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	//write the resources
+	return profile.writeProfileResources()
+
+}
+
+// updateExistingDefaultprofile migrates the old default profile to profile_old
+func updateExistingDefaultprofile(path string) error {
+	profile, err := NewConfigProfile(path, DefaultProfile)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+	meta, err := profile.metadata()
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+	meta.Name = "default_old"
+	err = profile.updateMetadata(meta)
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+	err = os.Rename(profile.ProfilePath, filepath.Join(profile.ProfilePath+"_old"))
+	if err != nil {
+		return errors.Wrap(err, "error moving default profile")
+	}
+
+	return nil
+}
+
+// CreateProfileFromDefaultLocation creates an existing profile from the default elastic-package config dir
+func CreateProfileFromDefaultLocation(profileName string, from string) error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+
+	return CreateProfileFrom(loc.StackDir(), profileName, from)
+}
+
+// CreateProfileFrom creates a new profile by copying over an existing profile
+func CreateProfileFrom(elasticPackagePath string, newProfileName string, fromProfileName string) error {
+
+	fromProfile, err := LoadProfile(elasticPackagePath, fromProfileName)
+	if err != nil {
+		return errors.Wrapf(err, "error loading %s profile", fromProfileName)
+	}
+
+	newProfile, err := NewConfigProfile(elasticPackagePath, newProfileName)
+	if err != nil {
+		return errors.Wrapf(err, "error creating %s profile", newProfileName)
+	}
+
+	newExists, err := newProfile.profileAlreadyExists()
+	if err != nil {
+		return errors.Wrapf(err, "error checking profile %s", newProfile.ProfilePath)
+	}
+	if newExists {
+		return errors.Errorf("profile %s already exists", newProfile.profileName)
+	}
+	os.Mkdir(newProfile.ProfilePath, 0755)
+
+	newProfile.updateFileBodies(fromProfile.configFiles)
+	return newProfile.writeProfileResources()
+
+}
+
+// LoadProfileFromDefaultLocation loads an existing profile from the default elastic-package config dir
+func LoadProfileFromDefaultLocation(profileName string) (*ConfigProfile, error) {
+
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding stack dir location")
+	}
+
+	return LoadProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfileFromDefaultLocation deletes a profile from the default elastic-package config dir
+func DeleteProfileFromDefaultLocation(profileName string) error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return DeleteProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfile deletes a given config profile.
+func DeleteProfile(elasticPackagePath string, profileName string) error {
```

Comment:
Why are `DeleteProfileFromDefaultLocation` and `DeleteProfile` both exposed?

Reply:
I think `DeleteProfile` could be unexported?